tree/unit.go

```diff
@@ -0,0 +1,79 @@
+package tree
+
+var UnitSize = 2
+
+type Unit byte
+
+func FirstNonPrefix(basekey []Unit, otherKey []Unit) Unit {
```

Comment:
`// Precondition: len(otherKey) > len(baseKey)`

Reply:
Now takes that into account.
```
        if len(smaller) > len(larger) {
		smaller = baseKey
		larger = otherKey
	}

	return larger[len(smaller)]
```